controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java

```diff
@@ -189,17 +189,17 @@ public String getScopeName() {
      */
     @Override
     public CompletableFuture<Segment> getSegment(final int number) {
-        return verifyLegalState(() -> getSegmentRow(number));
+        return verifyLegalState(getSegmentRow(number));
```

Comment:
this will run the code inside the future without verifying the state.. we pass a supplier so that only if the state is legal do we run the future from the supplier.